### ISSUE TITLE
`no-unbound-methods` `allowTypes` doesn't work for non-writable Signals

### ISSUE THREAD

**Issue by JasonWeinzierl** [JasonWeinzierl/eslint-plugin-rxjs-x#352]
#211 attempted to fix #209 by adding an `allowTypes` option that defaults to `['Signal']`.

However, this doesn't appear to work in practice, unless you add `'WritableSignal'` to the allowed types.  The unit tests seem to be incorrect, in that they don't perfectly match the current Angular type definition for `Signal`.

```ts
declare const SIGNAL: unique symbol;
type Signal<T> = (() => T) & {
  [SIGNAL]: unknown;
}
```

Swapping in the above definition to the "valid" unit test for `allowTypes` causes test failures.

Additionally, the "invalid" unit test seems incorrect in that it doesn't consider the default value of `allowTypes`.
